pygmt/clib/session.py

```diff
@@ -1242,7 +1242,7 @@ def virtualfile_from_grid(self, grid, registration="GMT_GRID_PIXEL_REG"):
         gmt_grid = self.create_data(
             family,
             geometry,
-            mode="GMT_CONTAINER_ONLY",
+            mode="GMT_CONTAINER_ONLY|GMT_GRID_IS_GEO",
```

Comment:
No. I didn't mean this change. Users can pass both Cartesian and geographic grids to GMT. We need to use `GMT_CONTAINER_ONLY` or `GMT_CONTAINER_ONLY|GMT_GRID_IS_GEO` for different grid type.

Reply:
Ah sorry, I thought it 'knew' automatically what the grid type was. Should actually get some tests in for this.

> BTW, `GMT_CONTAINER_ONLY` is equivalent to `GMT_CONTAINER_ONLY|GMT_GRID_IS_CARTESIAN`.

Ok, good to know.

Reply:
Just for reference, this was added in GMT 5.4. See:
- https://docs.generic-mapping-tools.org/dev/changes.html?highlight=gmt_grid_is_geo#api-changes
- https://docs.generic-mapping-tools.org/dev/api.html?highlight=gmt_grid_is_geo#exporting-a-data-set
- https://github.com/GenericMappingTools/gmt/commit/b6d0af653c7b5f33d6af172f1ec4ff6be9d4eed3.